File: src/main/java/com/example/storage/exception/handler/GlobalExceptionHandler.java
package com.example.storage.exception.handler;

import com.example.storage.exception.EmployeeNotFoundException;
import com.example.storage.exception.EquipmentNotAvailableException;
import com.example.storage.exception.EquipmentNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(EmployeeNotFoundException.class)
    public ResponseEntity<String> handleEmployeeNotFoundException(EmployeeNotFoundException e) {
        return new ResponseEntity<>(e.getMessage(), HttpStatus.UNAUTHORIZED);
    }

    @ExceptionHandler(EquipmentNotAvailableException.class)
    public ResponseEntity<String> handleEquipmentNotAvailableException(EquipmentNotAvailableException e) {
        return new ResponseEntity<>(e.getMessage(), HttpStatus.CONFLICT);
    }

    @ExceptionHandler(EquipmentNotFoundException.class)
    public ResponseEntity<String> handleEquipmentNotFoundException(EquipmentNotFoundException e) {
        return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
    }

    @ExceptionHandler(Exception.class)
    public ResponseEntity<String> handleGenericException(Exception e) {
        return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
    }
}

File: src/main/kotlin/ru/nto/storage/exception/handler/GlobalExceptionHandler.kt
package ru.nto.storage.exception.handler

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import ru.nto.storage.exception.EmployeeNotFoundException
import ru.nto.storage.exception.EquipmentNotAvailableException
import ru.nto.storage.exception.EquipmentNotFoundException

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(EmployeeNotFoundException::class)
    fun handleEmployeeNotFoundException(e: EmployeeNotFoundException): ResponseEntity<String> {
        return ResponseEntity(e.message, HttpStatus.UNAUTHORIZED)
    }

    @ExceptionHandler(EquipmentNotAvailableException::class)
    fun handleEquipmentNotAvailableException(e: EquipmentNotAvailableException): ResponseEntity<String> {
        return ResponseEntity(e.message, HttpStatus.CONFLICT)
    }

    @ExceptionHandler(EquipmentNotFoundException::class)
    fun handleEquipmentNotFoundException(e: EquipmentNotFoundException): ResponseEntity<String> {
        return ResponseEntity(e.message, HttpStatus.BAD_REQUEST)
    }

    @ExceptionHandler(Exception::class)
    fun handleGenericException(e: Exception): ResponseEntity<String> {
        return ResponseEntity(e.message, HttpStatus.BAD_REQUEST)
    }
}
